client/my-sites/post-type-list/index.jsx

```diff
@@ -54,22 +55,27 @@ class PostTypeList extends Component {
 		lastPageToRequest: PropTypes.number,
 	};
 
-	constructor() {
-		super( ...arguments );
+	constructor( props ) {
```

Comment:
Why don't you like the `super( ...arguments )` approach?

Reply:
There is nothing wrong with it, `super( props )` just seems to be used more often and is described this way in [React docs](https://reactjs.org/docs/react-component.html#constructor).

And if we didn't need to access `this.props` in the constructor, we wouldn't need to pass anything at all and call just `super()`. See [this discussion thread](https://discuss.reactjs.org/t/should-we-include-the-props-parameter-to-class-constructors-when-declaring-components-using-es6-classes/2781/2).

I also heard that using the `arguments` object in certain ways can prevent the JS engine from optimizing the code, so I try to avoid it.

Reply:
> I also heard that using the arguments object in certain ways can prevent the JS engine from optimizing the code

If you can find it, we can promote one style over another. Mostly the idea is to avoid using constructor when possible, and we've been allowing both usages. The ...args is from the default constructor

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/constructor

Reply:
Generally, modifying `arguments` or passing this object as parameter to another function is very suspicious.

`super(...arguments)` gets transpiled by Babel into: 
```js
DerivedClass.__proto__.apply(this, arguments);
````
so there is potential to be slow. Or maybe this particular usage is perfectly OK -- I'm unable to tell without being very familiar with internals of V8 and other engines.

Here is a list of common causes why V8 gives up optimizations: https://github.com/vhf/v8-bailout-reasons#assignment-to-parameter-in-arguments-object

And I also found the talk where I first heard about `arguments` as optimization killer: https://youtu.be/M1FBosB5tjM?t=30m57s (it's from Munich Node User Group meeting earlier this year and the presenter works at Google as V8 perf engineer)

My takeaway is to avoid using `arguments` whenever possible and use the equivalent ES6 features instead.

Reply:
p2 it ;)